script/multicallV2/DeployScript.sol

```diff
@@ -0,0 +1,291 @@
+
+    function validateMarkets() public {
+        ConfigMarketV2[] memory markets = getConfigMarket();
+
+        if (markets.length != configVariables.amountOfNewMarkets) {
+            console.log("markets.length", markets.length);
+            console.log(
+                "configVariables.amountOfNewMarkets",
+                configVariables.amountOfNewMarkets
+            );
+            revert("markets.length != configVariables.amountOfNewMarkets");
+        }
+        for (uint256 i = 0; i < markets.length; ++i) {
+            ConfigMarketV2 memory market = markets[i];
+            address controller = getController(market.isGenericController);
+            address depositAsset = getDepositAsset(market.depositAsset);
+            uint256 strkePrice = stringToUint(market.strikePrice);
+            if (!factory.controllers(controller)) {
+                console2.log("Controller not whitelisted", controller);
+                revert("Controller not whitelisted");
+            }
+            uint256 previewMarketID = factory.getMarketId(
+                market.token,
+                strkePrice,
+                depositAsset
+            );
+            address vault = factory.marketIdToVaults(previewMarketID, 0);
+            if (vault != address(0)) {
+                console2.log("Market already deployed", previewMarketID);
+                console2.log(
+                    "Market: ",
+                    market.token,
+                    strkePrice,
+                    depositAsset
+                );
+                revert("Market already deployed");
+            }
+        }
+    }
+
+    function deployEpochs() public {
+        // ConfigAddressesV2 memory addresses = getConfigAddresses(false);
+        // IERC20(0x5D59e5837F7e5d0F710178Eda34d9eCF069B36D2).approve(0x16B8004f5440f95D3347e55776066032a701c0E8, type(uint256).max);
+        ConfigEpochWithEmission[] memory epochs = getConfigEpochs();
+        MultiCaller.EpochConfig[]
+            memory _epochConfig = new MultiCaller.EpochConfig[](epochs.length);
+        MultiCaller.KeeperConfig[]
+            memory _keeperConfig = new MultiCaller.KeeperConfig[](
+                epochs.length
+            );
+
+        for (uint256 i = 0; i < epochs.length; ++i) {
+            ConfigEpochWithEmission memory epoch = epochs[i];
+            address depositAsset = getDepositAsset(epoch.depositAsset);
+            uint256 strikePrice = stringToUint(epoch.strikePrice);
+            uint256 marketId = factory.getMarketId(
+                epoch.token,
+                strikePrice,
+                depositAsset
+            );
+
+            // (uint256 epochId, address[2] memory vaults)
+            _epochConfig[i] = MultiCaller.EpochConfig(
+                marketId,
+                epoch.epochBegin,
+                epoch.epochEnd,
+                500,
+                stringToUint(epoch.premiumEmissions),
+                stringToUint(epoch.collatEmissions)
+            );
+
```

Comment:
@0xnyxos initialising the multical contract and actually calling it is missing here. I guess extending the ConfigAddressesV2 struct in the helper for a multical address would be a good solution.